src/Radium/Element.hs

```diff
@@ -195,64 +205,125 @@ configExceptions = Map.fromList [ (24, [2, 8, 13, 1])
 
 
 -- How many electrons are in each subshell
 subshellMaxElectrons :: [Int]
-subshellMaxElectrons = [2, 6, 10, 14, 18]
+subshellMaxElectrons = fmap (*2) subshellOrbitals
+
+subshellOrbitals :: [Int]
+subshellOrbitals = [1, 3, 5, 7, 9]
+
+
+-- Turn atom into the next excited state
+excite :: [(Int, Int, Int)] -> Maybe [(Int, Int, Int)]
+excite conf = fmap removeEmptyShells excConf
+  where
+    highestPaired :: [(Int, Int, Int)] -> Maybe Int
+    highestPaired = listToMaybe . reverse . findIndices (\(_, s, e) -> (subshellOrbitals !! (s - 1)) < e)
+
+    lowestFree :: [(Int, Int, Int)] -> Maybe Int
+    lowestFree = findIndex (\(_, s, e) -> (subshellOrbitals !! (s - 1)) > e)
+
+    increase (l, s, e) = (l, s, e + 1)
+    decrease (l, s, e) = (l, s, e - 1)
+
+    expConf = appendEmptyShells conf
+
+    excConf = case (highestPaired &&& lowestFree) expConf of
+      (Nothing, _)     -> Nothing
+      (_, Nothing)     -> Nothing
+      (Just h, Just l) -> if (conf !! h) ^._1 == (expConf !! l) ^._1
+                          then
+                            Just
+                              . toList
+                              . update l (increase $ expConf !! l)
+                              . update h (decrease $ expConf !! h)
+                              $ fromList expConf
+                          else
+                            Nothing
+
+-- Generate all possible electron configs (unexcited and excited) with given number of electrons
+possibleElectronConfigs :: Int -> [[(Int, Int, Int)]]
+possibleElectronConfigs e = fillShells e : unfoldr helper (fillShells e)
+  where
+    helper x = do
+      res <- excite x
+      return (res, res)
 
 -- Generate possible config for given maximum number of shells.
--- The order is based on Aufbau principle. 
+-- The order is based on Aufbau principle.
 -- http://en.wikipedia.org/wiki/Aufbau_principle
 shellConfigGen :: Int -> [(Int, Int)]
-shellConfigGen n = [(i,m-i) | m <- [2..n+n], i <- [((m+1) `div` 2)..m-1] ] 
+shellConfigGen n = [(i,m-i) | m <- [2..n+n], i <- [((m+1) `div` 2)..m-1] ]
+
+
+-- Appends sublayers which don't carry any electron in the ground state
+-- but could carry in excited state
+appendEmptyShells :: [(Int, Int, Int)] -> [(Int, Int, Int)]
+appendEmptyShells conf = conf ++ addSublayers
+  where
+    maxLayer = maximum (fmap (^._1) conf)
+    maxSublayer = maximum ((^._2) <$> filter ((==maxLayer) . (^._1)) conf)
+    addSublayers = fmap (\x -> (maxLayer, x, 0)) [(maxSublayer + 1) .. maxLayer]
+
+
+-- Do the oppsoite of the function above
+removeEmptyShells :: [(Int, Int, Int)] -> [(Int, Int, Int)]
+removeEmptyShells = filter ((/= 0) . (^._3))
+
 
 -- Fill shells with given number of electrons
 fillShells :: Int -> [( Int  -- Shell number
-                    , Int  -- Shell type
-                    , Int  -- #Electrons
-                      )]
+                      , Int  -- Shell type
+                      , Int  -- #Electrons
+                        )]
```

Comment:
Yes. Good idea. 
Do you want to do it?Or should I do it?

BTW there is one more thing. Since you added a lot of code to the Element.hs. I think it would be good to add your name to the copyright in the header. Or instead of copyright line add list of authors with your name.

Of course if it is ok with you?